java.lang.IllegalArgumentException: Service not registered: de.appplant.cordova.plugin.background.BackgroundMode

Randomly, and after having called setEnabled(true) successfully, when trying to call setEnabled(false), the following error occures:


```E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.acilim.sdx, PID: 13147
    java.lang.RuntimeException: Unable to resume activity {com.acilim.sdx/com.acilim.sdx.MainActivity}: java.lang.IllegalArgumentException: Service not registered: de.appplant.cordova.plugin.background.BackgroundMode$1@2f638ce
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3785)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3817)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6692)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalArgumentException: Service not registered: de.appplant.cordova.plugin.background.BackgroundMode$1@2f638ce
        at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1562)
        at android.app.ContextImpl.unbindService(ContextImpl.java:1692)
        at android.content.ContextWrapper.unbindService(ContextWrapper.java:717)
        at de.appplant.cordova.plugin.background.BackgroundMode.stopService(BackgroundMode.java:286)
        at de.appplant.cordova.plugin.background.BackgroundMode.onResume(BackgroundMode.java:172)
        at org.apache.cordova.PluginManager.onResume(PluginManager.java:264)
        at org.apache.cordova.CordovaWebViewImpl.handleResume(CordovaWebViewImpl.java:452)
        at org.apache.cordova.CordovaActivity.onResume(CordovaActivity.java:277)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1413)
        at android.app.Activity.performResume(Activity.java:7292)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3777)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3817) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6692) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
I/Process: Sending signal. PID: 13147 SIG: 9
Disconnected from the target VM, address: 'localhost:51019', transport: 'socket'
```

A possible solution is:

```
private void stopService()
    {
        Activity context = cordova.getActivity();
        Intent intent    = new Intent(context, ForegroundService.class);

        if (!isBind) return;

        try {
          fireEvent(Event.DEACTIVATE, null);
          context.unbindService(connection);
          context.stopService(intent); 
        }
        catch (java.lang.IllegalArgumentException) { }

        isBind = false;
    }
```

--- a/src/android/BackgroundMode.java
+++ b/src/android/BackgroundMode.java
@@ -278,14 +278,17 @@ public class BackgroundMode extends CordovaPlugin {
     private void stopService()
     {
         Activity context = cordova.getActivity();
-        Intent intent    = new Intent(context, ForegroundService.class);
+        Intent intent = new Intent(context, ForegroundService.class);
 
         if (!isBind) return;
 
-        fireEvent(Event.DEACTIVATE, null);
-        context.unbindService(connection);
-        context.stopService(intent);
-
+        try{
+            fireEvent(Event.DEACTIVATE, null);
+            context.unbindService(connection);
+            context.stopService(intent);
+        }catch(IllegalStateException e){
+            fireEvent(Event.FAILURE, String.format("'%s'", e.getMessage()));
+        }
         isBind = false;
     }
 

